src/cow-react/modules/limitOrders/hooks/useSetupLimitOrdersState.ts

```diff
@@ -21,17 +24,16 @@ export function useSetupLimitOrdersState() {
   useEffect(() => {
     if (shouldSkipUpdate) return
 
-    const newState: LimitOrdersState = chainIdWasChanged
+    const newState: Partial<LimitOrdersState> = chainIdWasChanged
       ? getDefaultLimitOrdersState(tradeStateFromUrl.chainId)
       : {
-          ...state,
           chainId: tradeStateFromUrl.chainId,
           recipient: tradeStateFromUrl.recipient || state.recipient,
           inputCurrencyId: tradeStateFromUrl.inputCurrencyId,
           outputCurrencyId: tradeStateFromUrl.outputCurrencyId,
         }
 
     console.log('UPDATE LIMIT ORDERS STATE:', newState)
-    setState(newState)
-  }, [setState, state, tradeStateFromUrl, shouldSkipUpdate, chainIdWasChanged])
+    updateLimitOrdersState(newState)
+  }, [updateLimitOrdersState, state, tradeStateFromUrl, shouldSkipUpdate, chainIdWasChanged])
```

Comment:
You are still depending on the full `state`. Would it work to depend only on state.recipient instead?

Reply:
Done